#============================================
#  File: 10b_24illumina_mirdeep2_mapper_output_summary_file.R
#  Directory Code:  /mnt/research/pigeqtl/analyses/microRNA/3_pcr_duplication_analysis/1_preprocess_24_illumina_samples/scripts
#  Date:  8/16/16
#  Description:  This code is used to create a summary of the reads mapped and unmapped for each library (n = 24)
#                Using sed on the command line to extract the correct lines from the OutputError file created after running
#                10_mirdeep2_24illumina_genome_mapper.sh, then used this file in combination with the 1_config_24illumina_mapper.txt file to
#                create the summary file.
#============================================
#  Input File Directory:  /mnt/research/pigeqtl/analyses/microRNA/OutputsErrors/3_preprocess_24illumina_fastq/
#						  /mnt/research/pigeqtl/analyses/microRNA/3_pcr_duplication_analysis/1_preprocess_24_illumina_samples/1_preprocess_24illumina_output/6_prepped_fasta_24illumina_output

#  Input File(s):   9_miRDeep2_24illumina_genome_mapper.o34362786
#                   1_config_24illumina_mapper.txt

#  Output File Directory: /mnt/research/pigeqtl/analyses/microRNA/3_pcr_duplication_analysis/1_preprocess_24_illumina_samples/8_mirdeep2_genome_mapper_24illumina_output/summary_files

#  Output File(s): 1_24illumina_mapper_outputerror_extraction.txt
#				   2_24illumina_mapper_output_summary.txt
#============================================
#  Module used: sed (command line), R/3.1.0
#============================================

#First, on command line, the following commands were used to extract the correct lines from the output file generated by mapping the 174 libraries to the reference genome (file = 10_174_library_miRDeep2_genome_mapper.o29209455)
cd /mnt/research/pigeqtl/analyses/microRNA/OutputsErrors/3_preprocess_24illumina_fastq
sed -n 171,196p 9_miRDeep2_24illumina_genome_mapper.o34362786 > /mnt/research/pigeqtl/analyses/microRNA/3_pcr_duplication_analysis/1_preprocess_24_illumina_samples/8_mirdeep2_genome_mapper_24illumina_output/summary_files/1_24illumina_mapper_outputerror_extraction.txt


#Then, in R:
R
#Set working directory:
setwd("/mnt/research/pigeqtl/analyses/microRNA/3_pcr_duplication_analysis/1_preprocess_24_illumina_samples/8_mirdeep2_genome_mapper_24illumina_output/summary_files")

mapstats<-read.table("1_24illumina_mapper_outputerror_extraction.txt")
head(mapstats)
#       V1        V2       V3       V4    V5    V6
# 1 total: 124199994 71843807 52356187 0.578 0.422
# 2   001:   2274890  1382762   892128 0.608 0.392
# 3   002:   4508239  2788651  1719588 0.619 0.381
# 4   003:   2746428  1678679  1067749 0.611 0.389
# 5   004:   4461712  2592870  1868842 0.581 0.419
# 6   005:   4442916  2598632  1844284 0.585 0.415

#Reassign column names as they were in the original text file:
colnames(mapstats)<-c("id","total","mapped","unmapped","percentmapped","percentunmapped")
head(mapstats)
#       id     total   mapped unmapped percentmapped percentunmapped
# 1 total: 124199994 71843807 52356187         0.578           0.422
# 2   001:   2274890  1382762   892128         0.608           0.392
# 3   002:   4508239  2788651  1719588         0.619           0.381
# 4   003:   2746428  1678679  1067749         0.611           0.389
# 5   004:   4461712  2592870  1868842         0.581           0.419
# 6   005:   4442916  2598632  1844284         0.585           0.415

dim(mapstats)
# [1] 25  6

#Remove "total" row:
mapstats<-mapstats[2:nrow(mapstats),]

head(mapstats)
#     id   total  mapped unmapped percentmapped percentunmapped
# 2 001: 2274890 1382762   892128         0.608           0.392
# 3 002: 4508239 2788651  1719588         0.619           0.381
# 4 003: 2746428 1678679  1067749         0.611           0.389
# 5 004: 4461712 2592870  1868842         0.581           0.419
# 6 005: 4442916 2598632  1844284         0.585           0.415
# 7 006: 5279044 3084461  2194583         0.584           0.416


mapstats$id<-as.character(sub(":", "", mapstats$id))

dim(mapstats)
# [1] 24  6

head(mapstats)
#    id   total  mapped unmapped percentmapped percentunmapped
# 2 001 2274890 1382762   892128         0.608           0.392
# 3 002 4508239 2788651  1719588         0.619           0.381
# 4 003 2746428 1678679  1067749         0.611           0.389
# 5 004 4461712 2592870  1868842         0.581           0.419
# 6 005 4442916 2598632  1844284         0.585           0.415
# 7 006 5279044 3084461  2194583         0.584           0.416

#Now, pull in the config_for_mapper file to bring in the animal ids:
config<-read.table("/mnt/research/pigeqtl/analyses/microRNA/3_pcr_duplication_analysis/1_preprocess_24_illumina_samples/1_preprocess_24illumina_output/6_prepped_fasta_24illumina_output/1_config_24illumina_mapper.txt")

head(config)
#                     V1 V2
# 1 1034_mapper_input.fa  1
# 2 1058_mapper_input.fa  2
# 3 1080_mapper_input.fa  3
# 4 1096_mapper_input.fa  4
# 5 1116_mapper_input.fa  5
# 6 1134_mapper_input.fa  6

#Convert the ids from config into characters, and maintain the 3-digit id:
config$V2<-as.character(sprintf("%03d",config$V2))

#Change column names:
colnames(config)<-c("inputfile", "id")
head(config)
#              inputfile  id
# 1 1034_mapper_input.fa 001
# 2 1058_mapper_input.fa 002
# 3 1080_mapper_input.fa 003
# 4 1096_mapper_input.fa 004
# 5 1116_mapper_input.fa 005
# 6 1134_mapper_input.fa 006

#Use "merge" function to bring these two datasets together, mergeing by "id" column:
mapsummary<-merge(config, mapstats, by = "id")
head(mapsummary)
#    id            inputfile   total  mapped unmapped percentmapped
# 1 001 1034_mapper_input.fa 2274890 1382762   892128         0.608
# 2 002 1058_mapper_input.fa 4508239 2788651  1719588         0.619
# 3 003 1080_mapper_input.fa 2746428 1678679  1067749         0.611
# 4 004 1096_mapper_input.fa 4461712 2592870  1868842         0.581
# 5 005 1116_mapper_input.fa 4442916 2598632  1844284         0.585
# 6 006 1134_mapper_input.fa 5279044 3084461  2194583         0.584
#   percentunmapped
# 1           0.392
# 2           0.381
# 3           0.389
# 4           0.419
# 5           0.415
# 6           0.416

#Check that no data was shuffled around in the merge:
sum(config$id != mapsummary$id)
# [1] 0
sum(config$inputfile != mapsummary$inputfile)
# [1] 0

#Looks good!

#Calculate the average mapped reads:
sum(mapsummary$mapped)
# [1] 71843807

sum(mapsummary$total)
# [1] 124199994

(sum(mapsummary$mapped))/(sum(mapsummary$total))
# [1] 0.5784526

mean(mapsummary$total)
# [1] 5175000
median(mapsummary$total)
# [1] 4452314
range(mapsummary$total)
# [1]  1402261 20300167
mean(mapsummary$mapped)
# [1] 2993492
median(mapsummary$mapped)
# [1] 2594332
range(mapsummary$mapped)
# [1]   827872 10548290

median(mapsummary$percentmapped)
# [1] 0.5875

mean(mapsummary$percentmapped)
# [1] 0.58825

summary(mapsummary$total)
 #    Min.  1st Qu.   Median     Mean  3rd Qu.     Max.
 # 1402000  3056000  4452000  5175000  5526000 20300000

summary(mapsummary$mapped)
  #   Min.  1st Qu.   Median     Mean  3rd Qu.     Max.
  # 827900  1682000  2594000  2993000  3203000 10550000

summary(mapsummary$percentmapped)
 #   Min. 1st Qu.  Median    Mean 3rd Qu.    Max.
 # 0.5200  0.5640  0.5875  0.5882  0.6165  0.6470

#Total reads mapped = 57.85%



#Save file:
write.table(mapsummary, file = "/mnt/research/pigeqtl/analyses/microRNA/3_pcr_duplication_analysis/1_preprocess_24_illumina_samples/8_mirdeep2_genome_mapper_24illumina_output/summary_files/2_24illumina_mapper_output_summary.txt", quote = FALSE, sep = " ", row.names = FALSE, col.names = TRUE)
q()
n






